build.sbt

```diff
@@ -1,66 +1,57 @@
-import sbt._
-import Keys._
-
 import com.typesafe.sbt.pgp.PgpKeys._
 import akka._
 
-name := "akka-http"
-
-val commonSettings =
-  Seq(
-    organization := "com.typesafe.akka",
-    organizationName := "Lightbend",
-    startYear := Some(2014),
-    test in assembly := {},
-    licenses := Seq("Apache License 2.0" -> url("http://opensource.org/licenses/Apache-2.0")),
-    scalaVersion := "2.11.8",
-    crossVersion := CrossVersion.binary,
-    scalacOptions ++= Seq(
-      "-deprecation",
-      "-encoding", "UTF-8", // yes, this is 2 args
-      "-unchecked",
-      "-Xlint",
-      // "-Yno-adapted-args", //akka-http heavily depends on adapted args and => Unit implicits break otherwise
-      "-Ywarn-dead-code"
-      // "-Xfuture" // breaks => Unit implicits
-    ),
-    testOptions += Tests.Argument(TestFrameworks.JUnit, "-q", "-v")) ++
-      Dependencies.Versions ++ akka.Formatting.formatSettings
-
-val dontPublishSettings = Seq(
-  publishSigned := (),
-  publish := (),
-  publishArtifact /* in Compile */ := false
-)
-
-lazy val parentSettings = Seq(
-  publishArtifact := false
-) ++ dontPublishSettings
+inThisBuild(Def.settings(
```

Comment:
I copied this style from Paradox. `inThisBuild` is a [month old](http://www.scala-sbt.org/0.13/docs/sbt-0.13-Tech-Previews.html#sbt+0.13.9) so it might be too early to call it "official".